### ISSUE TITLE
Automatically assigning states

### ISSUE THREAD

**Issue by psyllost** [hotNipi/node-red-contrib-ui-state-trail#3]
Hi,
Is it possible to get the states directly from the payload? The problem that I am phasing is that I have more than 50 states and it is a lot of manual work to add those to the node. States that are not added there, are not shown in the chart. Is there any workaround?

Thank you.

**Comment by hotNipi** [comment]
It can be done. But as for any dashboard widget on-fly configuration, it will not persist so you will need to re-send the configuration in any of event which makes node to reinitialize. (Like in case on deploy)

I'm afraid that I will not do the deep validation of input, just the explanation in info pages, how to format the input. 

```
var customstates = [
    {state:1,col:"#009933",t:"num",label:'ONE'},
    {state:2,col:"#999999",t:"num",label:'TWO'},
    {state:3,col:"#00FF99",t:"num",label:'THREE'}
]
msg.control = {states:customstates }
return msg;
```

It will be available with next public version.

Can I ask, what kind of data visualization you do with so many states? Just for to know how the widget is used so I can address future changes more clearly.

**Comment by psyllost** [comment]
Thank you for your prompt response.

I would like to visualise malfunctions occuring in different parts of a machine in the last 2 hours. Not all of the possible malfunctions will occur in this time span so it is also better to just show the labels of the states occuring in the current timeline chart in the legend instead of displaying all of them.

**Comment by hotNipi** [comment]
Thank you. Seems to me as very valid use case.
Requested functionality is already pushed to git (without documentation) but it takes at least a day or two until the next public version will be published to npm. Ongoing improvement in many areas needs more testing time. 

labels of the states occuring in the current timeline, should be also possible, but goes into plans after current milestone completed

**Comment by hotNipi** [comment]
Available with v0.2.6
